persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStore.java

```diff
@@ -322,6 +346,30 @@ public ConnectionFactory getConnectionFactory() {
       });
    }
 
+   private long runBatchAndNotify(List<PossibleExpirationNotification> possible, PreparedStatement batchDelete,
+         FlowableProcessor<MarshallableEntry<K, V>> flowable) throws SQLException {
+      long purgeAmount = 0;
+      int[] results = batchDelete.executeBatch();
+      if (possible != null) {
+         for (int i = 0; i < results.length; ++i) {
+            PossibleExpirationNotification notification = possible.get(i);
+            if (results[i] != Statement.EXECUTE_FAILED) {
+               Object key = ((TwoWayKey2StringMapper) key2StringMapper).getKeyMapping(notification.key);
+               MarshalledValue value = unmarshall(notification.is, marshaller);
+               flowable.onNext(marshalledEntryFactory.create(key, value));
+
+               purgeAmount++;
+            } else {
+               log.tracef("Unable to remove expired entry for key %s, most likely concurrent update", notification.key);
```

Comment:
```suggestion
               if (log.isTraceEnabled()) log.tracef("Unable to remove expired entry for key %s, most likely concurrent update", notification.key);
```

Reply:
:-1: The tracef already handles this, as there are no additional allocations. If we were storing the trace level in the object I can see using that, but we aren't in this class.